osquery/tables/system/windows/ntfs_acl_permissions.cpp

```diff
@@ -0,0 +1,227 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <windows.h>
+#include <AccCtrl.h>
+#include <Aclapi.h>
+#include <string>
+#include <map>
+#include <vector>
+
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+#include <boost/algorithm/string/join.hpp>
+#include <boost/filesystem.hpp>
+
+namespace alg = boost::algorithm;
+namespace fs = boost::filesystem;
+
+namespace osquery {
+namespace tables {
+
+// helper function to get access mode string
+std::string accessModeToStr(const _ACCESS_MODE& accessMode) {
+	switch (accessMode) {
+	case NOT_USED_ACCESS:
+		return "Not Used";
+	case GRANT_ACCESS:
+		return "Grant";
+	case SET_ACCESS:
+		return "Set";
+	case DENY_ACCESS:
+		return "Deny";
+	case REVOKE_ACCESS:
+		return "Revoke";
+	case SET_AUDIT_SUCCESS:
+		return "Set Audit Success";
+	case SET_AUDIT_FAILURE:
+		return "Set Audit Failure";
+	default:
+		return "Unknown";
+	}
+}
+
+// helper function to build access string from permission bit mask
+std::string accessPermsToStr(const unsigned long pmask) {
+	std::vector<std::string> permList;
+	std::string permStr = "";
+	const std::map<unsigned long, std::string> permVals = {
+		{ DELETE                         ,"Delete" },
+		{ READ_CONTROL                   ,"Read Control" },
+		{ WRITE_DAC                      ,"Write DAC" },
+		{ WRITE_OWNER                    ,"Write Owner" },
+		{ SYNCHRONIZE                    ,"Synchronize" },
+		{ STANDARD_RIGHTS_REQUIRED       ,"Std Rights Required" },
+		{ STANDARD_RIGHTS_ALL            ,"Std Rights All" },
+		{ SPECIFIC_RIGHTS_ALL            ,"Specific Rights All" },
+		{ ACCESS_SYSTEM_SECURITY         ,"Access System Security" },
+		{ MAXIMUM_ALLOWED                ,"Maximum Allowed" },
+		{ GENERIC_READ                   ,"Generic Read" },
+		{ GENERIC_WRITE                  ,"Generic Write" },
+		{ GENERIC_EXECUTE                ,"Generic Execute" },
+		{ GENERIC_ALL                    ,"Generic All" }
+	};
+
+	for (auto const& perm : permVals) {
+		if ((pmask & perm.first) != 0)
```

Comment:
Please put a `{}` around a body of condition, it prevent someone in a future from stupid mistake  :)

Reply:
👍